types.go

```diff
@@ -58,45 +58,67 @@ type Stats struct {
 	Indexes      map[string]StatsIndex `json:"indexes"`
 }
 
-// UpdateStatus is the status of an update.
-type UpdateStatus string
+// TaskStatus is the status of a task.
+type TaskStatus string
 
 const (
-	// UpdateStatusUnknown is the default UpdateStatus, should not exist
-	UpdateStatusUnknown UpdateStatus = "unknown"
-	// UpdateStatusEnqueued means the server know the update but didn't handle it yet
-	UpdateStatusEnqueued UpdateStatus = "enqueued"
-	// UpdateStatusProcessing means the server is processing the update and all went well
-	UpdateStatusProcessing UpdateStatus = "processing"
-	// UpdateStatusProcessed means the server has processed the update and all went well
-	UpdateStatusProcessed UpdateStatus = "processed"
-	// UpdateStatusFailed means the server has processed the update and an error has been reported
-	UpdateStatusFailed UpdateStatus = "failed"
+	// TaskStatusUnknown is the default TaskStatus, should not exist
+	TaskStatusUnknown TaskStatus = "unknown"
+	// TaskStatusEnqueued the task request has been received and will be processed soon
+	TaskStatusEnqueued TaskStatus = "enqueued"
+	// TaskStatusProcessing the task is being processed
+	TaskStatusProcessing TaskStatus = "processing"
+	// TaskStatusSucceeded the task has been successfully processed
+	TaskStatusSucceeded TaskStatus = "succeeded"
+	// TaskStatusFailed a failure occurred when processing the task, no changes were made to the database
+	TaskStatusFailed TaskStatus = "failed"
 )
 
-// Update indicate information about an update
-type Update struct {
-	Status      UpdateStatus `json:"status"`
-	UpdateID    int64        `json:"updateId"`
-	Type        Unknown      `json:"type"`
-	Error       string       `json:"error"`
-	EnqueuedAt  time.Time    `json:"enqueuedAt"`
-	ProcessedAt time.Time    `json:"processedAt"`
+// Task indicate information about a task is returned for asynchronous method
+//
+// Documentation: https://docs.meilisearch.com/learn/advanced/asynchronous_operations.html
+type Task struct {
+	Status     TaskStatus          `json:"status"`
+	UID        int64               `json:"uid"`
+	IndexUID   string              `json:"indexUid"`
+	Type       string              `json:"type"`
+	Error      meilisearchApiError `json:"error,omitempty"`
+	Duration   string              `json:"duration,omitempty"`
+	EnqueuedAt time.Time           `json:"enqueuedAt"`
+	StartedAt  time.Time           `json:"startedAt,omitempty"`
+	FinishedAt time.Time           `json:"finishedAt,omitempty"`
 }
 
-// AsyncUpdateID is returned for asynchronous method
-//
-// Documentation: https://docs.meilisearch.com/learn/advanced/asynchronous_updates.html
-type AsyncUpdateID struct {
-	UpdateID int64 `json:"updateId"`
+type ResultTask struct {
```

Comment:
It should be better to call it `Results` and make it generic as other get routes will return with a `results` fields. 

For example the keyRoute. 

Thus this is a duplicate with this: 

```
type ResultKey struct {
	Results []Key `json:"results"`
}
```

Is it possible to make it generic? Like this for example
```
type Results struct {
	Results []T `json:"results"`
}
```

and then use it like this for example: 
```
GetTasks() (resp *Results<Task>, err error)
```

Reply:
There is no `generic` in Go before the `v1.18` and we are in `1.17`.

Reply:
Good for me :)